packages/scalecube-microservice/src/MicroService/ServiceRegistry.ts

```diff
@@ -0,0 +1,70 @@
+import { ServiceEndPoint } from '../api/Service';
+import { generateIdentifier, isValidRawService } from '../helpers/utils';
+import { getServiceName } from '../helpers/utils';
+
+export const lookUp = ({ serviceRegistry, methodName }) => serviceRegistry[methodName];
+
+export const updateServiceRegistry = ({ serviceRegistry, rawService }) => {
+  if (isValidRawService(rawService)) {
+    const services = servicesFromRawService(rawService);
+    services.forEach((service) => addServiceToRegistry({ serviceRegistry, service }));
+  }
+
+  return { ...serviceRegistry };
+};
+
+const addServiceToRegistry = ({ serviceRegistry, service }) => {
+  const nameSpace = `${getServiceName(service)} ${service.methodName}`;
+  if (serviceRegistry[nameSpace]) {
+    serviceRegistry[nameSpace] = {
+      ...serviceRegistry[nameSpace],
+      ...serviceEndPoint(service),
+    };
+  } else {
+    serviceRegistry[nameSpace] = {
+      ...serviceEndPoint(service),
+    };
+  }
+
+  return { ...serviceRegistry };
```

Comment:
```javascript
const addServiceToRegistry = ({ serviceRegistry, service }) => {
  const nameSpace = `${getServiceName(service)} ${service.methodName}`;
  return { 
    ...serviceRegistry
    ...[nameSpace] ? { ...serviceEndPoint(service) } : { ...serviceEndPoint(service) }
   };
};
```

Reply:
actually we can do:

    serviceRegistry[nameSpace] = {
      ...(serviceRegistry[nameSpace] || {}),
      ...serviceEndPoint(service),
    };